### PR TITLE
Update resource names and lists for sensuctl dump and sensuctl prune

### LINKED ISSUE #2692: Use resource types defined in describe-type only
### Description
Commands such as `sensuctl dump` and `sensuctl prune` support the use of fully qualified names (ex. `core/v2.Check`) in addition to short names (ex. `check`, `checks`). To make these values more visible to users, `sensuctl describe-type all` can list the acceptable resource types. While this list does not contain every supported permutation of a short name (ex. kebab-case, singular, plural), it's best for us to limit the documentation to use the more reliable ways to express a resource type. We should update all examples of `sensuctl dump` and `sensuctl prune` so that their arguments are derived from the short name or fully qualified name provided by `sensuctl describe-type all`.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
- https://docs.sensu.io/sensu-go/latest/sensuctl/back-up-recover/#backdrop
- https://docs.sensu.io/sensu-go/latest/sensuctl/create-manage-resources/#sensuctl-prune

### Expected behavior
Only short name "synonyms" and fully qualified names from `sensuctl describe-type` are used in examples.

### Current behavior
Short name "synonyms" that are not listed in `sensuctl describe-type all` are sometimes used in examples (ex. `check` vs `checks`).

### Corrections needed
Validate the argument against the output of `sensuctl describe-type all`, update the argument if needed.

### Additional context
https://github.com/sensu/sensu-go/issues/3981 and https://github.com/sensu/sensu-go/issues/3981#issuecomment-681015034

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Corrects all sensuctl dump and sensuctl prune commands to use only the short names or fully qualified names listed when you run `sensuctl describe-type`.

Also updates the `sensuctl describe-type --all` list to reflect the response as displayed in terminal and explain how users can retrieve the current list rather than reproducing an edited list of resource types.

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2692

### CODE REVIEW

**Comment by hillaryfraley** [comment]
@nikkictl if this is correct and applies to 5.19, 5.20, and 5.21, I can propagate it to those versions.

File: content/sensu-go/6.0/sensuctl/back-up-recover.md [hunk 1]
```diff
@@ -129,24 +129,55 @@ When you export users, required password attributes are not included.
 You must add a [`password_hash`](../#generate-a-password-hash) or `password` to `users` resources before restoring them with the `sensuctl create` command.
 {{% /notice %}}
 
-## List types of supported resources
+## Sensuctl dump resource types
```
Comment (nikkictl) [inline content/sensu-go/6.0/sensuctl/back-up-recover.md]: I don't think you need to change this because these resource types are supported in other commands, ex. `sensuctl prune`.

Comment (nikkictl) [reply content/sensu-go/6.0/sensuctl/back-up-recover.md]:
```suggestion
## Supported resource types
```

File: content/sensu-go/6.0/sensuctl/back-up-recover.md [hunk 2]
```diff
@@ -129,24 +129,55 @@ When you export users, required password attributes are not included.
...
 **IMPORTANT**: The `sensuctl describe-type` command deprecates `sensuctl dump --types`.
 {{% /notice %}}
 
-Use `sensuctl describe-type` to list the types of supported resources.
-For example, to list all types:
+Use `sensuctl describe-type all` to retrieve the list of supported `sensuctl dump` resource types.
+
+{{% notice note %}}
+**NOTE**: The resource types with no short name listed are [commercial features](../../commercial/).
+{{% /notice %}}
```
Comment (nikkictl) [inline content/sensu-go/6.0/sensuctl/back-up-recover.md]: While this is technically true now, it's not necessarily true for the future (for example if a new OSS resource was created). I would rephrase this to something along the lines of `short names are only supported for core/v2 resources` since that's how it's implemented logically in our code.

File: content/sensu-go/6.0/sensuctl/back-up-recover.md [hunk 3]
```diff
@@ -156,50 +187,14 @@ sensuctl describe-type core/v2.CheckConfig,core/v2.EventFilter,core/v2.Handler
...
 ### Format the sensuctl describe-type response
 
 Add the `--format` flag to specify how the resources should be formatted in the `sensuctl describe-type` response.
 The default is unformatted, but you can specify either `wrapped-json` or `yaml`:
 
 {{< code shell >}}
-sensuctl describe-type core/v2.CheckConfig --format yaml
-sensuctl describe-type core/v2.CheckConfig --format wrapped-json
+sensuctl describe-type checks --format yaml
+sensuctl describe-type checks --format wrapped-json
```
Comment (nikkictl) [inline content/sensu-go/6.0/sensuctl/back-up-recover.md]: Let's encourage the use of the fully qualified name over the short name.

**Review by nikkictl — APPROVED** [review]
Thanks for applying my feedback of prioritizing the fully qualified names over the short names. I think it's okay to have _at least one_ example of a short name (short names look cleaner when you are listing multiple resource types). Otherwise, this LGTM!

**Comment by hillaryfraley** [comment]
Thank you for reviewing! We have an example in each doc where we show how to use the short names too.
